src/main/java/com/czertainly/core/service/impl/CertificateServiceImpl.java

```diff
@@ -1803,6 +1860,11 @@ public void switchRaProfile(SecuredUUID uuid, SecuredUUID raProfileUuid) throws
     public void updateCertificateGroups(SecuredUUID uuid, Set<UUID> groupUuids) throws NotFoundException {
         Certificate certificate = getCertificateEntityWithAssociations(uuid);
 
+        if (certificate.isArchived()) {
+            logger.info("Certificate with UUID {} is archived and its groups will not be updated.", uuid);
```

Comment:
Here is more suitable to throw Exception since it is user triggered operation that is not allowed. It should say it is not allowed to change groups for archived certificate.